spec/config/datafile_project_config_spec.rb

```diff
@@ -31,12 +31,22 @@
     it 'should initialize properties correctly upon creating project' do
       project_config = Optimizely::DatafileProjectConfig.new(config_body_JSON, logger, error_handler)
 
+      feature_flags_to_compare = config_body.fetch('featureFlags')
+      feature_flags_to_compare.each do |feature_flag|
+        feature_flag['variables'].each do |variable|
+          if variable['type'] == 'string' && variable['subType'] == 'json'
+            variable['type'] = 'json'
+            variable.delete('subType')
+          end
+        end
+      end
```

Comment:
Rather than putting the same manipulation code here. Can you directly access the two new variables that you have added and modify their type. Just a matter of preference, you may keep it this way.

Reply:
I have also noticed that, not sure why it has to be done in both places

Reply:
@oakbani @pawels-optimizely i changed it to only alter the ones that need to be altered.